src/Collection.php

```diff
@@ -188,7 +203,7 @@ public function __toString()
      * @see Aggregate::__construct() for supported options
      * @param array $pipeline Aggregation pipeline
      * @param array $options  Command options
-     * @return Cursor
+     * @return CursorInterface&Iterator
```

Comment:
This return type change is a breaking change when using static analysis tools. The user code can have type hint on the returned value.
But the returned value type will not change until the `codec` option is set. I don't think is can break any code.

Example of code that could break if a codec is provided:
```php
// Type hint
function process(Cursor $cursor) {
    // ...
}

/** @var Cursor is currently deducted from Collection::aggregate return type */
$result = $collection->aggregate([]);

process($result);
```

Reply:
Yes, that's a downside of the current change. Unfortunately, there is no way to change this without either removing `final` from our cursor implementation or making this small change. Since the behaviour is opt-in, I figured the change was benign enough to make in a minor release.